{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Notebook for the samples identification based on hashtags of Fetal Gut data from Fawkner-Corbett study\n",
    "- **Developed by:** Anna Maguza\n",
    "- **Place:** Wuerzburg Institute for System Immunology\n",
    "- **Date:** 13th March 2024"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import scanpy as sc\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import os\n",
    "import anndata\n",
    "\n",
    "from scipy import sparse"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set up the cells"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-----\n",
      "anndata     0.9.2\n",
      "scanpy      1.9.5\n",
      "-----\n",
      "PIL                         10.0.1\n",
      "anyio                       NA\n",
      "arrow                       1.3.0\n",
      "asttokens                   NA\n",
      "attr                        23.1.0\n",
      "attrs                       23.1.0\n",
      "babel                       2.13.0\n",
      "backcall                    0.2.0\n",
      "certifi                     2023.07.22\n",
      "cffi                        1.16.0\n",
      "charset_normalizer          3.3.0\n",
      "colorama                    0.4.6\n",
      "comm                        0.1.4\n",
      "cycler                      0.10.0\n",
      "cython_runtime              NA\n",
      "dateutil                    2.8.2\n",
      "debugpy                     1.8.0\n",
      "decorator                   5.1.1\n",
      "defusedxml                  0.7.1\n",
      "executing                   2.0.0\n",
      "fastjsonschema              NA\n",
      "fqdn                        NA\n",
      "h5py                        3.9.0\n",
      "idna                        3.4\n",
      "igraph                      0.11.2\n",
      "ipykernel                   6.25.2\n",
      "ipywidgets                  8.1.1\n",
      "isoduration                 NA\n",
      "jedi                        0.19.1\n",
      "jinja2                      3.1.2\n",
      "joblib                      1.3.2\n",
      "json5                       NA\n",
      "jsonpointer                 2.4\n",
      "jsonschema                  4.19.1\n",
      "jsonschema_specifications   NA\n",
      "jupyter_events              0.7.0\n",
      "jupyter_server              2.7.3\n",
      "jupyterlab_server           2.25.0\n",
      "kiwisolver                  1.4.5\n",
      "leidenalg                   0.10.1\n",
      "llvmlite                    0.40.1\n",
      "markupsafe                  2.1.3\n",
      "matplotlib                  3.8.0\n",
      "matplotlib_inline           0.1.6\n",
      "mpl_toolkits                NA\n",
      "natsort                     8.4.0\n",
      "nbformat                    5.9.2\n",
      "numba                       0.57.1\n",
      "numpy                       1.24.4\n",
      "overrides                   NA\n",
      "packaging                   23.2\n",
      "pandas                      2.1.1\n",
      "parso                       0.8.3\n",
      "pexpect                     4.8.0\n",
      "pickleshare                 0.7.5\n",
      "platformdirs                3.11.0\n",
      "prometheus_client           NA\n",
      "prompt_toolkit              3.0.39\n",
      "psutil                      5.9.5\n",
      "ptyprocess                  0.7.0\n",
      "pure_eval                   0.2.2\n",
      "pydev_ipython               NA\n",
      "pydevconsole                NA\n",
      "pydevd                      2.9.5\n",
      "pydevd_file_utils           NA\n",
      "pydevd_plugins              NA\n",
      "pydevd_tracing              NA\n",
      "pygments                    2.16.1\n",
      "pyparsing                   3.1.1\n",
      "pythonjsonlogger            NA\n",
      "pytz                        2023.3.post1\n",
      "referencing                 NA\n",
      "requests                    2.31.0\n",
      "rfc3339_validator           0.1.4\n",
      "rfc3986_validator           0.1.1\n",
      "rpds                        NA\n",
      "scipy                       1.11.3\n",
      "send2trash                  NA\n",
      "session_info                1.0.0\n",
      "six                         1.16.0\n",
      "sklearn                     1.3.1\n",
      "sniffio                     1.3.0\n",
      "stack_data                  0.6.3\n",
      "texttable                   1.7.0\n",
      "threadpoolctl               3.2.0\n",
      "tornado                     6.3.3\n",
      "traitlets                   5.11.2\n",
      "typing_extensions           NA\n",
      "uri_template                NA\n",
      "urllib3                     2.0.6\n",
      "vscode                      NA\n",
      "wcwidth                     0.2.8\n",
      "webcolors                   1.13\n",
      "websocket                   1.6.4\n",
      "yaml                        6.0.1\n",
      "zmq                         25.1.1\n",
      "zoneinfo                    NA\n",
      "zope                        NA\n",
      "-----\n",
      "IPython             8.16.1\n",
      "jupyter_client      8.3.1\n",
      "jupyter_core        5.3.2\n",
      "jupyterlab          4.0.6\n",
      "notebook            7.0.4\n",
      "-----\n",
      "Python 3.11.5 | packaged by conda-forge | (main, Aug 27 2023, 03:34:09) [GCC 12.3.0]\n",
      "Linux-6.6.14-100.fc38.x86_64-x86_64-with-glibc2.37\n",
      "-----\n",
      "Session information updated at 2024-03-13 14:51\n"
     ]
    }
   ],
   "source": [
    "sc.settings.verbosity = 3\n",
    "sc.logging.print_versions()\n",
    "sc.settings.set_figure_params(dpi = 180, color_map = 'magma_r', dpi_save = 300, vector_friendly = True, format = 'svg')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def X_is_raw(adata):\n",
    "    return np.array_equal(adata.X.sum(axis=0).astype(int), adata.X.sum(axis=0))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "+ Upload final anndata object"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/amaguza/miniforge3/envs/scanpy_env/lib/python3.11/site-packages/anndata/_core/anndata.py:1838: UserWarning: Observation names are not unique. To make them unique, call `.obs_names_make_unique`.\n",
      "  utils.warn_names_duplicates(\"obs\")\n",
      "/home/amaguza/miniforge3/envs/scanpy_env/lib/python3.11/site-packages/anndata/_core/anndata.py:1840: UserWarning: Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "  utils.warn_names_duplicates(\"var\")\n"
     ]
    }
   ],
   "source": [
    "final_anndata_path = '/mnt/LaCIE/annaM/gut_project/raw_data/Fawkner-Corbett_2021/scRNA_seq_data/anndata_files/Fawkner_Corbett_GEX_raw.h5ad'\n",
    "final_adata = sc.read_h5ad(final_anndata_path)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Upload sample info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_info = pd.read_csv('/mnt/LaCIE/annaM/gut_project/raw_data/Fawkner-Corbett_2021/scRNA_seq_data/Supplementary_Mendeley_data/1. Sample Overview-Table 1.csv', sep =';')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "+ Upload anndata objects with HTO info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/amaguza/miniforge3/envs/scanpy_env/lib/python3.11/site-packages/anndata/_core/anndata.py:1763: FutureWarning: The AnnData.concatenate method is deprecated in favour of the anndata.concat function. Please use anndata.concat instead.\n",
      "\n",
      "See the tutorial for concat at: https://anndata.readthedocs.io/en/latest/concatenation.html\n",
      "  warnings.warn(\n",
      "/home/amaguza/miniforge3/envs/scanpy_env/lib/python3.11/site-packages/anndata/_core/merge.py:217: FutureWarning: is_categorical_dtype is deprecated and will be removed in a future version. Use isinstance(dtype, CategoricalDtype) instead\n",
      "  if pd.api.types.is_categorical_dtype(dtype):\n",
      "/home/amaguza/miniforge3/envs/scanpy_env/lib/python3.11/site-packages/anndata/_core/anndata.py:121: ImplicitModificationWarning: Transforming to str index.\n",
      "  warnings.warn(\"Transforming to str index.\", ImplicitModificationWarning)\n",
      "/home/amaguza/miniforge3/envs/scanpy_env/lib/python3.11/site-packages/anndata/_core/anndata.py:1838: UserWarning: Observation names are not unique. To make them unique, call `.obs_names_make_unique`.\n",
      "  utils.warn_names_duplicates(\"obs\")\n",
      "/home/amaguza/miniforge3/envs/scanpy_env/lib/python3.11/site-packages/anndata/_core/merge.py:217: FutureWarning: is_categorical_dtype is deprecated and will be removed in a future version. Use isinstance(dtype, CategoricalDtype) instead\n",
      "  if pd.api.types.is_categorical_dtype(dtype):\n",
      "/home/amaguza/miniforge3/envs/scanpy_env/lib/python3.11/site-packages/anndata/_core/anndata.py:1838: UserWarning: Observation names are not unique. To make them unique, call `.obs_names_make_unique`.\n",
      "  utils.warn_names_duplicates(\"obs\")\n"
     ]
    }
   ],
   "source": [
    "# Path to the file with folder names\n",
    "folder_names_file = '/mnt/LaCIE/annaM/gut_project/raw_data/Fawkner-Corbett_2021/scRNA_seq_data/GSE158702_RAW_2/files_1.txt'\n",
    "\n",
    "# Base directory where the folders are located\n",
    "base_dir = '/mnt/LaCIE/annaM/gut_project/raw_data/Fawkner-Corbett_2021/scRNA_seq_data/GSE158702_RAW_2'\n",
    "\n",
    "# Initialize a list to store loaded AnnData objects\n",
    "adatas = []\n",
    "\n",
    "# Read folder names from the file\n",
    "with open(folder_names_file, 'r') as f:\n",
    "    folder_names = f.read().splitlines()\n",
    "\n",
    "# Loop through each folder, load the AnnData object, add the 'Folder' column, and append to the list\n",
    "for folder_name in folder_names:\n",
    "    folder_path = os.path.join(base_dir, folder_name)\n",
    "    for file in os.listdir(folder_path):\n",
    "        if file.endswith('.h5ad'):  # Assuming the AnnData object ends with .h5ad\n",
    "            file_path = os.path.join(folder_path, file)\n",
    "            adata = sc.read_h5ad(file_path)\n",
    "            adata.obs['Folder'] = folder_name  # Add new column with folder name\n",
    "            adatas.append(adata)\n",
    "            break  # Assuming there's only one AnnData object per folder\n",
    "\n",
    "# Concatenate all AnnData objects into one\n",
    "if adatas:\n",
    "    concatenated_adata = adatas[0].concatenate(adatas[1:], index_unique=None, join='outer')\n",
    "else:\n",
    "    concatenated_adata = None\n",
    "    print(\"No AnnData objects were loaded.\")\n",
    "\n",
    "# Save the concatenated AnnData object if needed\n",
    "if concatenated_adata is not None:\n",
    "    concatenated_adata.write('/mnt/LaCIE/annaM/gut_project/raw_data/Fawkner-Corbett_2021/scRNA_seq_data/GSE158702_RAW_2/HTO_counts_concatenated_adata.h5ad')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>HTO1</th>\n",
       "      <th>HTO2</th>\n",
       "      <th>HTO3</th>\n",
       "      <th>HTO4</th>\n",
       "      <th>HTO5</th>\n",
       "      <th>HTO6</th>\n",
       "      <th>HTO7</th>\n",
       "      <th>HTO8</th>\n",
       "      <th>HTO9</th>\n",
       "      <th>...</th>\n",
       "      <th>ADT1</th>\n",
       "      <th>ADT2</th>\n",
       "      <th>ADT3</th>\n",
       "      <th>ADT4</th>\n",
       "      <th>ADT5</th>\n",
       "      <th>ADT6</th>\n",
       "      <th>ADT7</th>\n",
       "      <th>ADT8</th>\n",
       "      <th>ADT9</th>\n",
       "      <th>batch</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>barcodes</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TACTGCCAGACCTTTG</th>\n",
       "      <td>TACTGCCAGACCTTTG</td>\n",
       "      <td>15.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CTGTGAACACTGAGGA</th>\n",
       "      <td>CTGTGAACACTGAGGA</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CAAGGGACAGGTGACA</th>\n",
       "      <td>CAAGGGACAGGTGACA</td>\n",
       "      <td>27.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>82.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GGGCCATAGTCACTAC</th>\n",
       "      <td>GGGCCATAGTCACTAC</td>\n",
       "      <td>22.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>2900.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>136.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AATTTCCCACTACCGG</th>\n",
       "      <td>AATTTCCCACTACCGG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTCGATTTCGTGTGAT</th>\n",
       "      <td>TTCGATTTCGTGTGAT</td>\n",
       "      <td>23.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTCTAGTAGAGTCACG</th>\n",
       "      <td>TTCTAGTAGAGTCACG</td>\n",
       "      <td>32.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>418.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TGAGGTTCACTATCCC</th>\n",
       "      <td>TGAGGTTCACTATCCC</td>\n",
       "      <td>31.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>85.0</td>\n",
       "      <td>3574.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTCAGGAGTTCAGGAG</th>\n",
       "      <td>TTCAGGAGTTCAGGAG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CTGAGGCTCGCGTTTC</th>\n",
       "      <td>CTGAGGCTCGCGTTTC</td>\n",
       "      <td>35.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>85.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2098481 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 0  HTO1  HTO2  HTO3    HTO4  HTO5   HTO6  \\\n",
       "barcodes                                                                    \n",
       "TACTGCCAGACCTTTG  TACTGCCAGACCTTTG  15.0   8.0   7.0     2.0   4.0   77.0   \n",
       "CTGTGAACACTGAGGA  CTGTGAACACTGAGGA   0.0   0.0   0.0     1.0   0.0    1.0   \n",
       "CAAGGGACAGGTGACA  CAAGGGACAGGTGACA  27.0  13.0  82.0     2.0   4.0    6.0   \n",
       "GGGCCATAGTCACTAC  GGGCCATAGTCACTAC  22.0  12.0  20.0  2900.0   4.0  136.0   \n",
       "AATTTCCCACTACCGG  AATTTCCCACTACCGG   1.0   0.0   0.0     0.0   0.0    0.0   \n",
       "...                            ...   ...   ...   ...     ...   ...    ...   \n",
       "TTCGATTTCGTGTGAT  TTCGATTTCGTGTGAT  23.0  51.0  46.0    79.0  10.0   37.0   \n",
       "TTCTAGTAGAGTCACG  TTCTAGTAGAGTCACG  32.0  57.0  67.0   418.0  33.0   53.0   \n",
       "TGAGGTTCACTATCCC  TGAGGTTCACTATCCC  31.0  45.0  85.0  3574.0  18.0   37.0   \n",
       "TTCAGGAGTTCAGGAG  TTCAGGAGTTCAGGAG   1.0   1.0   0.0     0.0   0.0    0.0   \n",
       "CTGAGGCTCGCGTTTC  CTGAGGCTCGCGTTTC  35.0  45.0  62.0    85.0  15.0   28.0   \n",
       "\n",
       "                  HTO7  HTO8  HTO9  ...  ADT1  ADT2  ADT3  ADT4  ADT5 ADT6  \\\n",
       "barcodes                            ...                                      \n",
       "TACTGCCAGACCTTTG   0.0   1.0   3.0  ...   NaN   NaN   NaN   NaN   NaN  NaN   \n",
       "CTGTGAACACTGAGGA   0.0   0.0   0.0  ...   NaN   NaN   NaN   NaN   NaN  NaN   \n",
       "CAAGGGACAGGTGACA   1.0   3.0   1.0  ...   NaN   NaN   NaN   NaN   NaN  NaN   \n",
       "GGGCCATAGTCACTAC   5.0   2.0   3.0  ...   NaN   NaN   NaN   NaN   NaN  NaN   \n",
       "AATTTCCCACTACCGG   0.0   0.0   0.0  ...   NaN   NaN   NaN   NaN   NaN  NaN   \n",
       "...                ...   ...   ...  ...   ...   ...   ...   ...   ...  ...   \n",
       "TTCGATTTCGTGTGAT   4.0  12.0   0.0  ...   NaN   NaN   NaN   NaN   NaN  NaN   \n",
       "TTCTAGTAGAGTCACG  11.0  17.0   0.0  ...   NaN   NaN   NaN   NaN   NaN  NaN   \n",
       "TGAGGTTCACTATCCC  11.0  24.0   0.0  ...   NaN   NaN   NaN   NaN   NaN  NaN   \n",
       "TTCAGGAGTTCAGGAG   0.0   0.0   0.0  ...   NaN   NaN   NaN   NaN   NaN  NaN   \n",
       "CTGAGGCTCGCGTTTC   7.0  10.0   0.0  ...   NaN   NaN   NaN   NaN   NaN  NaN   \n",
       "\n",
       "                 ADT7 ADT8 ADT9  batch  \n",
       "barcodes                                \n",
       "TACTGCCAGACCTTTG  NaN  NaN  NaN      0  \n",
       "CTGTGAACACTGAGGA  NaN  NaN  NaN      0  \n",
       "CAAGGGACAGGTGACA  NaN  NaN  NaN      0  \n",
       "GGGCCATAGTCACTAC  NaN  NaN  NaN      0  \n",
       "AATTTCCCACTACCGG  NaN  NaN  NaN      0  \n",
       "...               ...  ...  ...    ...  \n",
       "TTCGATTTCGTGTGAT  NaN  NaN  NaN     11  \n",
       "TTCTAGTAGAGTCACG  NaN  NaN  NaN     11  \n",
       "TGAGGTTCACTATCCC  NaN  NaN  NaN     11  \n",
       "TTCAGGAGTTCAGGAG  NaN  NaN  NaN     11  \n",
       "CTGAGGCTCGCGTTTC  NaN  NaN  NaN     11  \n",
       "\n",
       "[2098481 rows x 30 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatenated_adata.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "in_house_folders = ['GSM4808359_ADT1', 'GSM4808360_ADT2', 'GSM4808361_ADT3']\n",
    "\n",
    "concatenated_adata.obs['Tag_type'] = concatenated_adata.obs['Folder'].apply(lambda x: 'In-house' if x in in_house_folders else 'TotalSeq-tag')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(concatenated_adata.obs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# leave only columns 'most_likely_hypothesis', 'cluster_feature', 'negative_hypothesis_probability', 'singlet_hypothesis_probability', 'doublet_hypothesis_probability', 'Classification', 'Digestion_Condition', 'GEO_Accession', 'Folder'\n",
    "df = df[['most_likely_hypothesis', 'cluster_feature', 'negative_hypothesis_probability', 'singlet_hypothesis_probability', 'doublet_hypothesis_probability', 'Classification', 'Digestion_Condition', 'GEO_Accession', 'Folder', 'Tag_type']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "+ Change SRA Run table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "SraRunTable = pd.read_csv('/mnt/LaCIE/annaM/gut_project/raw_data/Fawkner-Corbett_2021/scRNA_seq_data/SraRunTable.txt', sep =',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Assuming SraRunTable is already loaded with 'post-conceptual_weeks' column containing values like '8,9'\n",
    "# First, replace ' and ' with ',' to standardize the separator, then ensure all separators are just a single comma\n",
    "SraRunTable['post-conceptual_weeks'] = SraRunTable['post-conceptual_weeks'].str.replace(' and ', ',')\n",
    "SraRunTable['post-conceptual_weeks'] = SraRunTable['post-conceptual_weeks'].str.replace(' PCW', '')\n",
    "SraRunTable['post-conceptual_weeks'] = SraRunTable['post-conceptual_weeks'].str.replace(' ', '')  # Remove any additional spaces\n",
    "\n",
    "# Ensure that the separator between numbers is uniformly a comma\n",
    "SraRunTable['post-conceptual_weeks'] = SraRunTable['post-conceptual_weeks'].apply(lambda x: ','.join(x.split(',')))\n",
    "\n",
    "# Now split based on ',' to ensure all values like '8,9' are separated\n",
    "SraRunTable['post-conceptual_weeks'] = SraRunTable['post-conceptual_weeks'].str.split(',')\n",
    "\n",
    "# Explode the DataFrame to create a new row for each post-conceptual_week value\n",
    "SraRunTable_exploded = SraRunTable.explode('post-conceptual_weeks')\n",
    "\n",
    "# Convert 'post-conceptual_weeks' to integers\n",
    "SraRunTable_exploded['post-conceptual_weeks'] = SraRunTable_exploded['post-conceptual_weeks'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>Assay Type</th>\n",
       "      <th>AvgSpotLen</th>\n",
       "      <th>Bases</th>\n",
       "      <th>BioProject</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Bytes</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>Center Name</th>\n",
       "      <th>Consent</th>\n",
       "      <th>...</th>\n",
       "      <th>LibrarySource</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Platform</th>\n",
       "      <th>post-conceptual_weeks</th>\n",
       "      <th>ReleaseDate</th>\n",
       "      <th>create_date</th>\n",
       "      <th>version</th>\n",
       "      <th>Sample Name</th>\n",
       "      <th>source_name</th>\n",
       "      <th>SRA Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR12735687</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>300</td>\n",
       "      <td>1476707400</td>\n",
       "      <td>PRJNA666217</td>\n",
       "      <td>SAMN16284014</td>\n",
       "      <td>560528715</td>\n",
       "      <td>EPCAM+ intestinal epithelial cells</td>\n",
       "      <td>GEO</td>\n",
       "      <td>public</td>\n",
       "      <td>...</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>8</td>\n",
       "      <td>2021-01-04T00:00:00Z</td>\n",
       "      <td>2020-09-28T19:21:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM4808339</td>\n",
       "      <td>Small and large fetal intestine</td>\n",
       "      <td>SRP285688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR12735687</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>300</td>\n",
       "      <td>1476707400</td>\n",
       "      <td>PRJNA666217</td>\n",
       "      <td>SAMN16284014</td>\n",
       "      <td>560528715</td>\n",
       "      <td>EPCAM+ intestinal epithelial cells</td>\n",
       "      <td>GEO</td>\n",
       "      <td>public</td>\n",
       "      <td>...</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>9</td>\n",
       "      <td>2021-01-04T00:00:00Z</td>\n",
       "      <td>2020-09-28T19:21:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM4808339</td>\n",
       "      <td>Small and large fetal intestine</td>\n",
       "      <td>SRP285688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR12735687</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>300</td>\n",
       "      <td>1476707400</td>\n",
       "      <td>PRJNA666217</td>\n",
       "      <td>SAMN16284014</td>\n",
       "      <td>560528715</td>\n",
       "      <td>EPCAM+ intestinal epithelial cells</td>\n",
       "      <td>GEO</td>\n",
       "      <td>public</td>\n",
       "      <td>...</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>13</td>\n",
       "      <td>2021-01-04T00:00:00Z</td>\n",
       "      <td>2020-09-28T19:21:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM4808339</td>\n",
       "      <td>Small and large fetal intestine</td>\n",
       "      <td>SRP285688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR12735687</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>300</td>\n",
       "      <td>1476707400</td>\n",
       "      <td>PRJNA666217</td>\n",
       "      <td>SAMN16284014</td>\n",
       "      <td>560528715</td>\n",
       "      <td>EPCAM+ intestinal epithelial cells</td>\n",
       "      <td>GEO</td>\n",
       "      <td>public</td>\n",
       "      <td>...</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>20</td>\n",
       "      <td>2021-01-04T00:00:00Z</td>\n",
       "      <td>2020-09-28T19:21:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM4808339</td>\n",
       "      <td>Small and large fetal intestine</td>\n",
       "      <td>SRP285688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR12735688</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>300</td>\n",
       "      <td>903216900</td>\n",
       "      <td>PRJNA666217</td>\n",
       "      <td>SAMN16284014</td>\n",
       "      <td>342383725</td>\n",
       "      <td>EPCAM+ intestinal epithelial cells</td>\n",
       "      <td>GEO</td>\n",
       "      <td>public</td>\n",
       "      <td>...</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>8</td>\n",
       "      <td>2021-01-04T00:00:00Z</td>\n",
       "      <td>2020-09-28T19:21:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM4808339</td>\n",
       "      <td>Small and large fetal intestine</td>\n",
       "      <td>SRP285688</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Run Assay Type  AvgSpotLen       Bases   BioProject     BioSample  \\\n",
       "0  SRR12735687    RNA-Seq         300  1476707400  PRJNA666217  SAMN16284014   \n",
       "0  SRR12735687    RNA-Seq         300  1476707400  PRJNA666217  SAMN16284014   \n",
       "0  SRR12735687    RNA-Seq         300  1476707400  PRJNA666217  SAMN16284014   \n",
       "0  SRR12735687    RNA-Seq         300  1476707400  PRJNA666217  SAMN16284014   \n",
       "1  SRR12735688    RNA-Seq         300   903216900  PRJNA666217  SAMN16284014   \n",
       "\n",
       "       Bytes                           cell_type Center Name Consent  ...  \\\n",
       "0  560528715  EPCAM+ intestinal epithelial cells         GEO  public  ...   \n",
       "0  560528715  EPCAM+ intestinal epithelial cells         GEO  public  ...   \n",
       "0  560528715  EPCAM+ intestinal epithelial cells         GEO  public  ...   \n",
       "0  560528715  EPCAM+ intestinal epithelial cells         GEO  public  ...   \n",
       "1  342383725  EPCAM+ intestinal epithelial cells         GEO  public  ...   \n",
       "\n",
       "    LibrarySource      Organism  Platform post-conceptual_weeks  \\\n",
       "0  TRANSCRIPTOMIC  Homo sapiens  ILLUMINA                     8   \n",
       "0  TRANSCRIPTOMIC  Homo sapiens  ILLUMINA                     9   \n",
       "0  TRANSCRIPTOMIC  Homo sapiens  ILLUMINA                    13   \n",
       "0  TRANSCRIPTOMIC  Homo sapiens  ILLUMINA                    20   \n",
       "1  TRANSCRIPTOMIC  Homo sapiens  ILLUMINA                     8   \n",
       "\n",
       "            ReleaseDate           create_date version Sample Name  \\\n",
       "0  2021-01-04T00:00:00Z  2020-09-28T19:21:00Z       1  GSM4808339   \n",
       "0  2021-01-04T00:00:00Z  2020-09-28T19:21:00Z       1  GSM4808339   \n",
       "0  2021-01-04T00:00:00Z  2020-09-28T19:21:00Z       1  GSM4808339   \n",
       "0  2021-01-04T00:00:00Z  2020-09-28T19:21:00Z       1  GSM4808339   \n",
       "1  2021-01-04T00:00:00Z  2020-09-28T19:21:00Z       1  GSM4808339   \n",
       "\n",
       "                       source_name  SRA Study  \n",
       "0  Small and large fetal intestine  SRP285688  \n",
       "0  Small and large fetal intestine  SRP285688  \n",
       "0  Small and large fetal intestine  SRP285688  \n",
       "0  Small and large fetal intestine  SRP285688  \n",
       "1  Small and large fetal intestine  SRP285688  \n",
       "\n",
       "[5 rows x 28 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SraRunTable_exploded.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Map cell_type to Digestion Condition for compatibility\n",
    "digestion_map = {\n",
    "    'EPCAM+ intestinal epithelial cells': 'EpCAM+',\n",
    "    'EPCAM- intestinal stromal cells': 'EpCAM-',\n",
    "    'intestinal stromal and epithelial cells': 'Bulk digest'\n",
    "}\n",
    "SraRunTable_exploded['Digestion Condition'] = SraRunTable_exploded['cell_type'].map(digestion_map)\n",
    "\n",
    "# Convert 'post-conceptual_weeks' in SraRunTable_exploded and 'PCW' in sample_info to the same type for accurate merging\n",
    "SraRunTable_exploded['post-conceptual_weeks'] = SraRunTable_exploded['post-conceptual_weeks'].astype(str)\n",
    "sample_info['PCW'] = sample_info['PCW'].astype(str)\n",
    "\n",
    "# Merge on 'post-conceptual_weeks'/'PCW' and 'Digestion Condition'\n",
    "merged_info = pd.merge(sample_info, SraRunTable_exploded, left_on=['PCW', 'Digestion Condition'], right_on=['post-conceptual_weeks', 'Digestion Condition'], how='left')\n",
    "\n",
    "# Aggregate GEO_Accession values for each sample into a list (if there are multiple GEO_Accession values per sample)\n",
    "aggregated_info = merged_info.groupby(['PCW', 'Digestion Condition'], as_index=False).agg({'Sample Name': lambda x: ', '.join(x.dropna().unique())})\n",
    "\n",
    "# Merge the aggregated_info back into sample_info to add the 'GEO_Accession' column\n",
    "final_sample_info = pd.merge(sample_info, aggregated_info, on=['PCW', 'Digestion Condition'], how='left')\n",
    "\n",
    "# Rename the 'GEO_Accession' column in final_sample_info if needed\n",
    "final_sample_info.rename(columns={'GEO_Accession_x': 'GEO_Accession'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample  ID</th>\n",
       "      <th>Method</th>\n",
       "      <th>PCW</th>\n",
       "      <th>Gender / Genotype</th>\n",
       "      <th>Location</th>\n",
       "      <th>Digestion Condition</th>\n",
       "      <th>Run number</th>\n",
       "      <th>10x reaction number</th>\n",
       "      <th>HTO number (TotalSeq, Biolegend)</th>\n",
       "      <th>HTO number (In house, Stoeckius et al, 2018)</th>\n",
       "      <th>HTO pool D70X_s</th>\n",
       "      <th>ADT Pool RPI</th>\n",
       "      <th>Sample Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAE</td>\n",
       "      <td>scRNA-seq</td>\n",
       "      <td>19</td>\n",
       "      <td>XX</td>\n",
       "      <td>Distal colon</td>\n",
       "      <td>EpCAM+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAE</td>\n",
       "      <td>scRNA-seq</td>\n",
       "      <td>19</td>\n",
       "      <td>XX</td>\n",
       "      <td>Distal colon</td>\n",
       "      <td>EpCAM-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAF</td>\n",
       "      <td>scRNA-seq</td>\n",
       "      <td>12</td>\n",
       "      <td>XY</td>\n",
       "      <td>Hindgut</td>\n",
       "      <td>EpCAM+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM4808341, GSM4808351, GSM4808361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAF</td>\n",
       "      <td>scRNA-seq</td>\n",
       "      <td>12</td>\n",
       "      <td>XY</td>\n",
       "      <td>Hindgut</td>\n",
       "      <td>EpCAM-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM4808344, GSM4808354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAB</td>\n",
       "      <td>scRNA-seq</td>\n",
       "      <td>17</td>\n",
       "      <td>XX</td>\n",
       "      <td>Distal colon</td>\n",
       "      <td>EpCAM+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GSM4808340, GSM4808350, GSM4808360</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Sample  ID     Method PCW Gender / Genotype      Location  \\\n",
       "0        AAE  scRNA-seq  19                XX  Distal colon   \n",
       "1        AAE  scRNA-seq  19                XX  Distal colon   \n",
       "2        AAF  scRNA-seq  12                XY       Hindgut   \n",
       "3        AAF  scRNA-seq  12                XY       Hindgut   \n",
       "4        AAB  scRNA-seq  17                XX  Distal colon   \n",
       "\n",
       "  Digestion Condition  Run number  10x reaction number  \\\n",
       "0              EpCAM+         1.0                  1.0   \n",
       "1              EpCAM-         1.0                  1.0   \n",
       "2              EpCAM+         1.0                  1.0   \n",
       "3              EpCAM-         1.0                  1.0   \n",
       "4              EpCAM+         1.0                  1.0   \n",
       "\n",
       "   HTO number (TotalSeq, Biolegend)  \\\n",
       "0                               1.0   \n",
       "1                               2.0   \n",
       "2                               3.0   \n",
       "3                               4.0   \n",
       "4                               5.0   \n",
       "\n",
       "   HTO number (In house, Stoeckius et al, 2018)  HTO pool D70X_s   \\\n",
       "0                                           NaN               1.0   \n",
       "1                                           NaN               1.0   \n",
       "2                                           NaN               1.0   \n",
       "3                                           NaN               1.0   \n",
       "4                                           NaN               1.0   \n",
       "\n",
       "   ADT Pool RPI                         Sample Name  \n",
       "0           NaN                                      \n",
       "1           NaN                                      \n",
       "2           NaN  GSM4808341, GSM4808351, GSM4808361  \n",
       "3           NaN              GSM4808344, GSM4808354  \n",
       "4           NaN  GSM4808340, GSM4808350, GSM4808360  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_sample_info.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Scanpy",
   "language": "python",
   "name": "scnapy-server"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
