{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create anndata object for Haber, 2017 LGR5+ stem cells\n",
    "\n",
    "- **Developed by:** Anna Maguza\n",
    "- **Affilation:** Faculty of Medicine, Würzburg University\n",
    "- **Date of creation:** 19th August 2024\n",
    "- **Last modified date:** 20th August 2024"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "+ Import packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import scanpy as sc\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import anndata as ad"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "+ Load datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0610005C13Rik</th>\n",
       "      <th>0610007N19Rik</th>\n",
       "      <th>0610007P14Rik</th>\n",
       "      <th>0610008F07Rik</th>\n",
       "      <th>0610009B14Rik</th>\n",
       "      <th>0610009B22Rik</th>\n",
       "      <th>0610009D07Rik</th>\n",
       "      <th>0610009L18Rik</th>\n",
       "      <th>0610009O20Rik</th>\n",
       "      <th>0610010B08Rik</th>\n",
       "      <th>...</th>\n",
       "      <th>Zxda</th>\n",
       "      <th>Zxdb</th>\n",
       "      <th>Zxdc</th>\n",
       "      <th>Zyg11a</th>\n",
       "      <th>Zyg11b</th>\n",
       "      <th>Zyx</th>\n",
       "      <th>Zzef1</th>\n",
       "      <th>Zzz3</th>\n",
       "      <th>a</th>\n",
       "      <th>l7Rn6</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S634_Lgr5GFP_M7_GFPLow_Enterocyte</th>\n",
       "      <td>103.90</td>\n",
       "      <td>21.93</td>\n",
       "      <td>150.61</td>\n",
       "      <td>8.6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>59.52</td>\n",
       "      <td>33.52</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.08</td>\n",
       "      <td>0.64</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.67</td>\n",
       "      <td>1.04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.82</td>\n",
       "      <td>11.72</td>\n",
       "      <td>55.49</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S745_Lgr5GFP_M7_GFPLow_Stem</th>\n",
       "      <td>0.00</td>\n",
       "      <td>70.47</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.22</td>\n",
       "      <td>51.87</td>\n",
       "      <td>0.0</td>\n",
       "      <td>54.82</td>\n",
       "      <td>45.11</td>\n",
       "      <td>...</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>7.72</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.01</td>\n",
       "      <td>3.77</td>\n",
       "      <td>4.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>53.27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S856_Lgr5GFP_M7_GFPLow_Stem</th>\n",
       "      <td>8.99</td>\n",
       "      <td>20.84</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>76.65</td>\n",
       "      <td>219.94</td>\n",
       "      <td>0.0</td>\n",
       "      <td>57.28</td>\n",
       "      <td>7.79</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.00</td>\n",
       "      <td>14.46</td>\n",
       "      <td>0.56</td>\n",
       "      <td>0.0</td>\n",
       "      <td>57.09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S967_Lgr5GFP_M7_GFPLow_Goblet</th>\n",
       "      <td>0.00</td>\n",
       "      <td>67.15</td>\n",
       "      <td>212.32</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>64.97</td>\n",
       "      <td>45.24</td>\n",
       "      <td>0.0</td>\n",
       "      <td>135.89</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>33.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>37.72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S1078_Lgr5GFP_M7_GFPLow_Goblet</th>\n",
       "      <td>50.94</td>\n",
       "      <td>36.25</td>\n",
       "      <td>104.89</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>63.68</td>\n",
       "      <td>93.14</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.47</td>\n",
       "      <td>0.00</td>\n",
       "      <td>16.56</td>\n",
       "      <td>15.39</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 20108 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   0610005C13Rik  0610007N19Rik  \\\n",
       "S634_Lgr5GFP_M7_GFPLow_Enterocyte         103.90          21.93   \n",
       "S745_Lgr5GFP_M7_GFPLow_Stem                 0.00          70.47   \n",
       "S856_Lgr5GFP_M7_GFPLow_Stem                 8.99          20.84   \n",
       "S967_Lgr5GFP_M7_GFPLow_Goblet               0.00          67.15   \n",
       "S1078_Lgr5GFP_M7_GFPLow_Goblet             50.94          36.25   \n",
       "\n",
       "                                   0610007P14Rik  0610008F07Rik  \\\n",
       "S634_Lgr5GFP_M7_GFPLow_Enterocyte         150.61            8.6   \n",
       "S745_Lgr5GFP_M7_GFPLow_Stem                 0.00            0.0   \n",
       "S856_Lgr5GFP_M7_GFPLow_Stem                 0.00            0.0   \n",
       "S967_Lgr5GFP_M7_GFPLow_Goblet             212.32            0.0   \n",
       "S1078_Lgr5GFP_M7_GFPLow_Goblet            104.89            0.0   \n",
       "\n",
       "                                   0610009B14Rik  0610009B22Rik  \\\n",
       "S634_Lgr5GFP_M7_GFPLow_Enterocyte            0.0          59.52   \n",
       "S745_Lgr5GFP_M7_GFPLow_Stem                  0.0          10.22   \n",
       "S856_Lgr5GFP_M7_GFPLow_Stem                  0.0          76.65   \n",
       "S967_Lgr5GFP_M7_GFPLow_Goblet                0.0          64.97   \n",
       "S1078_Lgr5GFP_M7_GFPLow_Goblet               0.0          63.68   \n",
       "\n",
       "                                   0610009D07Rik  0610009L18Rik  \\\n",
       "S634_Lgr5GFP_M7_GFPLow_Enterocyte          33.52            0.0   \n",
       "S745_Lgr5GFP_M7_GFPLow_Stem                51.87            0.0   \n",
       "S856_Lgr5GFP_M7_GFPLow_Stem               219.94            0.0   \n",
       "S967_Lgr5GFP_M7_GFPLow_Goblet              45.24            0.0   \n",
       "S1078_Lgr5GFP_M7_GFPLow_Goblet             93.14            0.0   \n",
       "\n",
       "                                   0610009O20Rik  0610010B08Rik  ...  Zxda  \\\n",
       "S634_Lgr5GFP_M7_GFPLow_Enterocyte           1.08           0.64  ...  0.00   \n",
       "S745_Lgr5GFP_M7_GFPLow_Stem                54.82          45.11  ...  0.22   \n",
       "S856_Lgr5GFP_M7_GFPLow_Stem                57.28           7.79  ...  0.00   \n",
       "S967_Lgr5GFP_M7_GFPLow_Goblet             135.89           0.00  ...  0.00   \n",
       "S1078_Lgr5GFP_M7_GFPLow_Goblet              1.05           0.00  ...  0.39   \n",
       "\n",
       "                                   Zxdb  Zxdc  Zyg11a  Zyg11b    Zyx  Zzef1  \\\n",
       "S634_Lgr5GFP_M7_GFPLow_Enterocyte  1.67  1.04     0.0    3.82  11.72  55.49   \n",
       "S745_Lgr5GFP_M7_GFPLow_Stem        0.00  7.72     0.0   12.01   3.77   4.03   \n",
       "S856_Lgr5GFP_M7_GFPLow_Stem        0.00  0.00     0.0    0.26   0.00  14.46   \n",
       "S967_Lgr5GFP_M7_GFPLow_Goblet      0.00  0.00     0.0    5.01   0.00  33.99   \n",
       "S1078_Lgr5GFP_M7_GFPLow_Goblet     0.00  0.00     0.0    0.47   0.00  16.56   \n",
       "\n",
       "                                    Zzz3    a  l7Rn6  \n",
       "S634_Lgr5GFP_M7_GFPLow_Enterocyte   0.00  0.0   5.99  \n",
       "S745_Lgr5GFP_M7_GFPLow_Stem         0.00  0.0  53.27  \n",
       "S856_Lgr5GFP_M7_GFPLow_Stem         0.56  0.0  57.09  \n",
       "S967_Lgr5GFP_M7_GFPLow_Goblet       0.00  0.0  37.72  \n",
       "S1078_Lgr5GFP_M7_GFPLow_Goblet     15.39  0.0   0.00  \n",
       "\n",
       "[5 rows x 20108 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AtlasFullLength_TPM = pd.read_csv('/../../../gut_project/raw_fastq_files/Haber_2017/GSE92332/GSE92332_AtlasFullLength_TPM.txt', sep='\\t', index_col=0)\n",
    "AtlasFullLength_TPM = AtlasFullLength_TPM.T\n",
    "AtlasFullLength_TPM.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "AtlasFullLength_TPM['barcode'] = AtlasFullLength_TPM.index.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "AtlasFullLength_TPM[['barcode', 'Gene_marker', 'Donor_ID', 'GFP_intensity', 'Cell_Type']] = AtlasFullLength_TPM['barcode'].str.split('_', expand=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0610005C13Rik</th>\n",
       "      <th>0610007N19Rik</th>\n",
       "      <th>0610007P14Rik</th>\n",
       "      <th>0610008F07Rik</th>\n",
       "      <th>0610009B14Rik</th>\n",
       "      <th>0610009B22Rik</th>\n",
       "      <th>0610009D07Rik</th>\n",
       "      <th>0610009L18Rik</th>\n",
       "      <th>0610009O20Rik</th>\n",
       "      <th>0610010B08Rik</th>\n",
       "      <th>...</th>\n",
       "      <th>Zyx</th>\n",
       "      <th>Zzef1</th>\n",
       "      <th>Zzz3</th>\n",
       "      <th>a</th>\n",
       "      <th>l7Rn6</th>\n",
       "      <th>barcode</th>\n",
       "      <th>Gene_marker</th>\n",
       "      <th>Donor_ID</th>\n",
       "      <th>GFP_intensity</th>\n",
       "      <th>Cell_Type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S634_Lgr5GFP_M7_GFPLow_Enterocyte</th>\n",
       "      <td>103.90</td>\n",
       "      <td>21.93</td>\n",
       "      <td>150.61</td>\n",
       "      <td>8.6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>59.52</td>\n",
       "      <td>33.52</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.08</td>\n",
       "      <td>0.64</td>\n",
       "      <td>...</td>\n",
       "      <td>11.72</td>\n",
       "      <td>55.49</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.99</td>\n",
       "      <td>S634</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Enterocyte</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S745_Lgr5GFP_M7_GFPLow_Stem</th>\n",
       "      <td>0.00</td>\n",
       "      <td>70.47</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.22</td>\n",
       "      <td>51.87</td>\n",
       "      <td>0.0</td>\n",
       "      <td>54.82</td>\n",
       "      <td>45.11</td>\n",
       "      <td>...</td>\n",
       "      <td>3.77</td>\n",
       "      <td>4.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>53.27</td>\n",
       "      <td>S745</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Stem</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S856_Lgr5GFP_M7_GFPLow_Stem</th>\n",
       "      <td>8.99</td>\n",
       "      <td>20.84</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>76.65</td>\n",
       "      <td>219.94</td>\n",
       "      <td>0.0</td>\n",
       "      <td>57.28</td>\n",
       "      <td>7.79</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>14.46</td>\n",
       "      <td>0.56</td>\n",
       "      <td>0.0</td>\n",
       "      <td>57.09</td>\n",
       "      <td>S856</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Stem</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S967_Lgr5GFP_M7_GFPLow_Goblet</th>\n",
       "      <td>0.00</td>\n",
       "      <td>67.15</td>\n",
       "      <td>212.32</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>64.97</td>\n",
       "      <td>45.24</td>\n",
       "      <td>0.0</td>\n",
       "      <td>135.89</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>33.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>37.72</td>\n",
       "      <td>S967</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Goblet</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S1078_Lgr5GFP_M7_GFPLow_Goblet</th>\n",
       "      <td>50.94</td>\n",
       "      <td>36.25</td>\n",
       "      <td>104.89</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>63.68</td>\n",
       "      <td>93.14</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>16.56</td>\n",
       "      <td>15.39</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>S1078</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Goblet</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 20113 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   0610005C13Rik  0610007N19Rik  \\\n",
       "S634_Lgr5GFP_M7_GFPLow_Enterocyte         103.90          21.93   \n",
       "S745_Lgr5GFP_M7_GFPLow_Stem                 0.00          70.47   \n",
       "S856_Lgr5GFP_M7_GFPLow_Stem                 8.99          20.84   \n",
       "S967_Lgr5GFP_M7_GFPLow_Goblet               0.00          67.15   \n",
       "S1078_Lgr5GFP_M7_GFPLow_Goblet             50.94          36.25   \n",
       "\n",
       "                                   0610007P14Rik  0610008F07Rik  \\\n",
       "S634_Lgr5GFP_M7_GFPLow_Enterocyte         150.61            8.6   \n",
       "S745_Lgr5GFP_M7_GFPLow_Stem                 0.00            0.0   \n",
       "S856_Lgr5GFP_M7_GFPLow_Stem                 0.00            0.0   \n",
       "S967_Lgr5GFP_M7_GFPLow_Goblet             212.32            0.0   \n",
       "S1078_Lgr5GFP_M7_GFPLow_Goblet            104.89            0.0   \n",
       "\n",
       "                                   0610009B14Rik  0610009B22Rik  \\\n",
       "S634_Lgr5GFP_M7_GFPLow_Enterocyte            0.0          59.52   \n",
       "S745_Lgr5GFP_M7_GFPLow_Stem                  0.0          10.22   \n",
       "S856_Lgr5GFP_M7_GFPLow_Stem                  0.0          76.65   \n",
       "S967_Lgr5GFP_M7_GFPLow_Goblet                0.0          64.97   \n",
       "S1078_Lgr5GFP_M7_GFPLow_Goblet               0.0          63.68   \n",
       "\n",
       "                                   0610009D07Rik  0610009L18Rik  \\\n",
       "S634_Lgr5GFP_M7_GFPLow_Enterocyte          33.52            0.0   \n",
       "S745_Lgr5GFP_M7_GFPLow_Stem                51.87            0.0   \n",
       "S856_Lgr5GFP_M7_GFPLow_Stem               219.94            0.0   \n",
       "S967_Lgr5GFP_M7_GFPLow_Goblet              45.24            0.0   \n",
       "S1078_Lgr5GFP_M7_GFPLow_Goblet             93.14            0.0   \n",
       "\n",
       "                                   0610009O20Rik  0610010B08Rik  ...    Zyx  \\\n",
       "S634_Lgr5GFP_M7_GFPLow_Enterocyte           1.08           0.64  ...  11.72   \n",
       "S745_Lgr5GFP_M7_GFPLow_Stem                54.82          45.11  ...   3.77   \n",
       "S856_Lgr5GFP_M7_GFPLow_Stem                57.28           7.79  ...   0.00   \n",
       "S967_Lgr5GFP_M7_GFPLow_Goblet             135.89           0.00  ...   0.00   \n",
       "S1078_Lgr5GFP_M7_GFPLow_Goblet              1.05           0.00  ...   0.00   \n",
       "\n",
       "                                   Zzef1   Zzz3    a  l7Rn6  barcode  \\\n",
       "S634_Lgr5GFP_M7_GFPLow_Enterocyte  55.49   0.00  0.0   5.99     S634   \n",
       "S745_Lgr5GFP_M7_GFPLow_Stem         4.03   0.00  0.0  53.27     S745   \n",
       "S856_Lgr5GFP_M7_GFPLow_Stem        14.46   0.56  0.0  57.09     S856   \n",
       "S967_Lgr5GFP_M7_GFPLow_Goblet      33.99   0.00  0.0  37.72     S967   \n",
       "S1078_Lgr5GFP_M7_GFPLow_Goblet     16.56  15.39  0.0   0.00    S1078   \n",
       "\n",
       "                                   Gene_marker  Donor_ID  GFP_intensity  \\\n",
       "S634_Lgr5GFP_M7_GFPLow_Enterocyte      Lgr5GFP        M7         GFPLow   \n",
       "S745_Lgr5GFP_M7_GFPLow_Stem            Lgr5GFP        M7         GFPLow   \n",
       "S856_Lgr5GFP_M7_GFPLow_Stem            Lgr5GFP        M7         GFPLow   \n",
       "S967_Lgr5GFP_M7_GFPLow_Goblet          Lgr5GFP        M7         GFPLow   \n",
       "S1078_Lgr5GFP_M7_GFPLow_Goblet         Lgr5GFP        M7         GFPLow   \n",
       "\n",
       "                                    Cell_Type  \n",
       "S634_Lgr5GFP_M7_GFPLow_Enterocyte  Enterocyte  \n",
       "S745_Lgr5GFP_M7_GFPLow_Stem              Stem  \n",
       "S856_Lgr5GFP_M7_GFPLow_Stem              Stem  \n",
       "S967_Lgr5GFP_M7_GFPLow_Goblet          Goblet  \n",
       "S1078_Lgr5GFP_M7_GFPLow_Goblet         Goblet  \n",
       "\n",
       "[5 rows x 20113 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AtlasFullLength_TPM.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "X = AtlasFullLength_TPM.iloc[:, 1:20108] \n",
    "var = pd.DataFrame(index=AtlasFullLength_TPM.columns[1:20108])\n",
    "obs = AtlasFullLength_TPM.iloc[:, 20108:20114]\n",
    "\n",
    "adata = ad.AnnData(X=X.values, var=var, obs=obs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata.write('/../../../gut_project/raw_fastq_files/Haber_2017/Haber_2017_Smartseq_LGR5_FACS_data.h5ad')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "+ Add human orthologue genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata_copy = adata.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_mouse_orthologues = pd.read_csv('/../../../ensembl_data/mouse_to_human_orthologues_ensembl.txt', sep='\\t', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "85408"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "duplicates = human_mouse_orthologues['Gene name'].duplicated()\n",
    "num_duplicates = duplicates.sum()\n",
    "num_duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_mouse_orthologues_cleaned = human_mouse_orthologues.loc[\n",
    "    human_mouse_orthologues.groupby('Gene name')['%id. target Human gene identical to query gene'].idxmax()\n",
    "]\n",
    "human_mouse_orthologues_cleaned = human_mouse_orthologues_cleaned.reset_index(drop=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_mouse_orthologues_cleaned = human_mouse_orthologues_cleaned.drop_duplicates(subset='Gene name')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "14814"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "intersect = np.intersect1d(human_mouse_orthologues_cleaned['Gene name'], adata.var.index)\n",
    "len(intersect)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/amaguza/miniforge3/envs/scanpy_env/lib/python3.11/site-packages/anndata/_core/anndata.py:1113: FutureWarning: is_categorical_dtype is deprecated and will be removed in a future version. Use isinstance(dtype, CategoricalDtype) instead\n",
      "  if not is_categorical_dtype(df_full[k]):\n"
     ]
    }
   ],
   "source": [
    "human_mouse_orthologues_cleaned = human_mouse_orthologues_cleaned.loc[human_mouse_orthologues_cleaned['Gene name'].isin(intersect)]\n",
    "adata = adata[:, intersect]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_460533/2987893276.py:1: ImplicitModificationWarning: Trying to modify attribute `.var` of view, initializing view as actual.\n",
      "  adata.var['original_mouse_gene_name'] = adata.var.index.copy()\n"
     ]
    }
   ],
   "source": [
    "adata.var['original_mouse_gene_name'] = adata.var.index.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/amaguza/miniforge3/envs/scanpy_env/lib/python3.11/site-packages/anndata/_core/anndata.py:788: UserWarning: \n",
      "AnnData expects .var.index to contain strings, but got values like:\n",
      "    [0, 14, 18, 24, 30]\n",
      "\n",
      "    Inferred to be: integer\n",
      "\n",
      "  value_idx = self._prep_dim_index(value.index, attr)\n"
     ]
    }
   ],
   "source": [
    "adata.var = adata.var.merge(human_mouse_orthologues_cleaned, left_index=True, right_on='Gene name')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata.var = adata.var.set_index('Human gene name')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata.var.index = adata.var.index.astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "adata.write('/../../../gut_project/Processed_data/Gut_data/Haber_2017_Smartseq/Haber_2017_Smartseq_LGR5_FACS_data_human_orthologues_genes.h5ad')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "+ Create anndata object from re-mapped data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "countmatrix = pd.read_csv('/../../../gut_project/raw_fastq_files/Haber_2017/counts_all.txt', \n",
    "                          sep='\\t', \n",
    "                          comment='#', \n",
    "                          skiprows=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "countmatrix.columns = countmatrix.columns.str.split('/').str[-1].str.replace('_Aligned.sortedByCoord.out.bam', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "countmatrix = countmatrix.set_index('Geneid')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "+ Go from SRR to cell name as in FACS data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/amaguza/miniforge3/envs/scanpy_env/lib/python3.11/site-packages/anndata/_core/anndata.py:1840: UserWarning: Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "  utils.warn_names_duplicates(\"var\")\n",
      "/home/amaguza/miniforge3/envs/scanpy_env/lib/python3.11/site-packages/anndata/_core/anndata.py:1113: FutureWarning: is_categorical_dtype is deprecated and will be removed in a future version. Use isinstance(dtype, CategoricalDtype) instead\n",
      "  if not is_categorical_dtype(df_full[k]):\n",
      "/tmp/ipykernel_17745/294400955.py:4: ImplicitModificationWarning: Trying to modify attribute `.obs` of view, initializing view as actual.\n",
      "  adata.obs['full_cell_description'] = adata.obs.index.copy()\n",
      "/home/amaguza/miniforge3/envs/scanpy_env/lib/python3.11/site-packages/anndata/_core/anndata.py:1840: UserWarning: Variable names are not unique. To make them unique, call `.var_names_make_unique`.\n",
      "  utils.warn_names_duplicates(\"var\")\n"
     ]
    }
   ],
   "source": [
    "adata = sc.read_h5ad('/../../../gut_project/Processed_data/Gut_data/Haber_2017_Smartseq/Haber_2017_Smartseq_LGR5_FACS_data_human_orthologues_genes.h5ad')\n",
    "adata = adata[adata.obs['Cell_Type'] == 'Stem']\n",
    "adata = adata[adata.obs['Gene_marker'] == 'Lgr5GFP']\n",
    "adata.obs['full_cell_description'] = adata.obs.index.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "dataframe = pd.read_csv('/../../../gut_project/raw_fastq_files/Haber_2017/sample_database.csv')\n",
    "dataframe = dataframe[dataframe['Title'].str.contains('Atlas WT single cell ')]\n",
    "dataframe['Sample'] = dataframe['Title'].str.split('Atlas WT single cell ').str[1]\n",
    "dataframe['Sample'] = 'S' + dataframe['Sample']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "dataframe = dataframe[dataframe['Sample'].isin(adata.obs['barcode'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Geneid</th>\n",
       "      <th>ENSMUSG00000102693.2</th>\n",
       "      <th>ENSMUSG00000064842.3</th>\n",
       "      <th>ENSMUSG00000051951.6</th>\n",
       "      <th>ENSMUSG00000102851.2</th>\n",
       "      <th>ENSMUSG00000103377.2</th>\n",
       "      <th>ENSMUSG00000104017.2</th>\n",
       "      <th>ENSMUSG00000103025.2</th>\n",
       "      <th>ENSMUSG00000089699.2</th>\n",
       "      <th>ENSMUSG00000103201.2</th>\n",
       "      <th>ENSMUSG00000103147.2</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSMUSG00000096550.2</th>\n",
       "      <th>ENSMUSG00000094172.2</th>\n",
       "      <th>ENSMUSG00000094887.2</th>\n",
       "      <th>ENSMUSG00000091585.3</th>\n",
       "      <th>ENSMUSG00000095763.2</th>\n",
       "      <th>ENSMUSG00000095523.2</th>\n",
       "      <th>ENSMUSG00000095475.2</th>\n",
       "      <th>ENSMUSG00000094855.2</th>\n",
       "      <th>ENSMUSG00000095019.2</th>\n",
       "      <th>ENSMUSG00000095041.8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SRX2418522</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418533</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>904</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418544</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418555</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418577</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418588</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418599</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418622</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418633</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418666</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 57186 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Geneid     ENSMUSG00000102693.2 ENSMUSG00000064842.3 ENSMUSG00000051951.6  \\\n",
       "SRX2418522                    0                    0                    0   \n",
       "SRX2418533                    0                    0                    0   \n",
       "SRX2418544                    0                    0                    0   \n",
       "SRX2418555                    0                    0                    0   \n",
       "SRX2418577                    0                    0                    0   \n",
       "SRX2418588                    0                    0                    0   \n",
       "SRX2418599                    0                    0                    0   \n",
       "SRX2418622                    0                    0                    0   \n",
       "SRX2418633                    0                    0                    0   \n",
       "SRX2418666                    0                    0                    0   \n",
       "\n",
       "Geneid     ENSMUSG00000102851.2 ENSMUSG00000103377.2 ENSMUSG00000104017.2  \\\n",
       "SRX2418522                    0                    0                    0   \n",
       "SRX2418533                    0                    0                    0   \n",
       "SRX2418544                    0                    0                    0   \n",
       "SRX2418555                    0                    0                    0   \n",
       "SRX2418577                    0                    0                    0   \n",
       "SRX2418588                    0                    0                    0   \n",
       "SRX2418599                    0                    0                    0   \n",
       "SRX2418622                    0                    0                    0   \n",
       "SRX2418633                    0                    0                    0   \n",
       "SRX2418666                    0                    0                    0   \n",
       "\n",
       "Geneid     ENSMUSG00000103025.2 ENSMUSG00000089699.2 ENSMUSG00000103201.2  \\\n",
       "SRX2418522                    0                    0                    0   \n",
       "SRX2418533                    0                    0                    0   \n",
       "SRX2418544                    0                    0                    0   \n",
       "SRX2418555                    0                    0                    0   \n",
       "SRX2418577                    0                    0                    0   \n",
       "SRX2418588                    0                    0                    0   \n",
       "SRX2418599                    0                    0                    0   \n",
       "SRX2418622                    0                    0                    0   \n",
       "SRX2418633                    0                    0                    0   \n",
       "SRX2418666                    0                    0                    0   \n",
       "\n",
       "Geneid     ENSMUSG00000103147.2  ... ENSMUSG00000096550.2  \\\n",
       "SRX2418522                    0  ...                    0   \n",
       "SRX2418533                    0  ...                    0   \n",
       "SRX2418544                    0  ...                    0   \n",
       "SRX2418555                    0  ...                    0   \n",
       "SRX2418577                    0  ...                    0   \n",
       "SRX2418588                    0  ...                    0   \n",
       "SRX2418599                    0  ...                    0   \n",
       "SRX2418622                    0  ...                    0   \n",
       "SRX2418633                    0  ...                    0   \n",
       "SRX2418666                    0  ...                    0   \n",
       "\n",
       "Geneid     ENSMUSG00000094172.2 ENSMUSG00000094887.2 ENSMUSG00000091585.3  \\\n",
       "SRX2418522                    0                    0                    0   \n",
       "SRX2418533                    0                    0                    0   \n",
       "SRX2418544                    0                    0                    0   \n",
       "SRX2418555                    0                    0                    0   \n",
       "SRX2418577                    0                    0                    0   \n",
       "SRX2418588                    0                    0                    0   \n",
       "SRX2418599                    0                    0                    0   \n",
       "SRX2418622                    0                    0                    0   \n",
       "SRX2418633                    0                    0                    0   \n",
       "SRX2418666                    0                    0                    0   \n",
       "\n",
       "Geneid     ENSMUSG00000095763.2 ENSMUSG00000095523.2 ENSMUSG00000095475.2  \\\n",
       "SRX2418522                    0                    0                    0   \n",
       "SRX2418533                    0                    0                    0   \n",
       "SRX2418544                    0                    0                    0   \n",
       "SRX2418555                    0                    0                    0   \n",
       "SRX2418577                    0                    0                    0   \n",
       "SRX2418588                    0                    0                    0   \n",
       "SRX2418599                    0                    0                    0   \n",
       "SRX2418622                    0                    0                    0   \n",
       "SRX2418633                    0                    0                    0   \n",
       "SRX2418666                    0                    0                    0   \n",
       "\n",
       "Geneid     ENSMUSG00000094855.2 ENSMUSG00000095019.2 ENSMUSG00000095041.8  \n",
       "SRX2418522                    0                    0                  105  \n",
       "SRX2418533                    0                    0                  904  \n",
       "SRX2418544                    0                    0                  354  \n",
       "SRX2418555                    0                    0                   43  \n",
       "SRX2418577                    0                    0                  138  \n",
       "SRX2418588                    0                    0                  502  \n",
       "SRX2418599                    0                    0                  136  \n",
       "SRX2418622                    0                    0                   18  \n",
       "SRX2418633                    0                    0                  529  \n",
       "SRX2418666                    0                    0                   78  \n",
       "\n",
       "[10 rows x 57186 columns]"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "countmatrix = countmatrix.T\n",
    "countmatrix = countmatrix.iloc[5:]\n",
    "countmatrix.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "sra_accession_df = dataframe[['Sample', 'SRA Accession']]\n",
    "adata.obs = adata.obs.merge(sra_accession_df, left_on='barcode', right_on='Sample', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>barcode</th>\n",
       "      <th>Gene_marker</th>\n",
       "      <th>Donor_ID</th>\n",
       "      <th>GFP_intensity</th>\n",
       "      <th>Cell_Type</th>\n",
       "      <th>full_cell_description</th>\n",
       "      <th>Sample</th>\n",
       "      <th>SRA Accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S745</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S745_Lgr5GFP_M7_GFPLow_Stem</td>\n",
       "      <td>S745</td>\n",
       "      <td>SRX2419245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S856</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S856_Lgr5GFP_M7_GFPLow_Stem</td>\n",
       "      <td>S856</td>\n",
       "      <td>SRX2419356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S222</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S222_Lgr5GFP_M7_GFPHigh_Stem</td>\n",
       "      <td>S222</td>\n",
       "      <td>SRX2418722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S333</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S333_Lgr5GFP_M7_GFPLow_Stem</td>\n",
       "      <td>S333</td>\n",
       "      <td>SRX2418833</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S589</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S589_Lgr5GFP_M7_GFPLow_Stem</td>\n",
       "      <td>S589</td>\n",
       "      <td>SRX2419089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>487</th>\n",
       "      <td>S423</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M5</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S423_Lgr5GFP_M5_GFPHigh_Stem</td>\n",
       "      <td>S423</td>\n",
       "      <td>SRX2418923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>488</th>\n",
       "      <td>S424</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M5</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S424_Lgr5GFP_M5_GFPHigh_Stem</td>\n",
       "      <td>S424</td>\n",
       "      <td>SRX2418924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>489</th>\n",
       "      <td>S425</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M5</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S425_Lgr5GFP_M5_GFPHigh_Stem</td>\n",
       "      <td>S425</td>\n",
       "      <td>SRX2418925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>490</th>\n",
       "      <td>S427</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M5</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S427_Lgr5GFP_M5_GFPHigh_Stem</td>\n",
       "      <td>S427</td>\n",
       "      <td>SRX2418927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>491</th>\n",
       "      <td>S429</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M5</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S429_Lgr5GFP_M5_GFPHigh_Stem</td>\n",
       "      <td>S429</td>\n",
       "      <td>SRX2418929</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>492 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    barcode Gene_marker Donor_ID GFP_intensity Cell_Type  \\\n",
       "0      S745     Lgr5GFP       M7        GFPLow      Stem   \n",
       "1      S856     Lgr5GFP       M7        GFPLow      Stem   \n",
       "2      S222     Lgr5GFP       M7       GFPHigh      Stem   \n",
       "3      S333     Lgr5GFP       M7        GFPLow      Stem   \n",
       "4      S589     Lgr5GFP       M7        GFPLow      Stem   \n",
       "..      ...         ...      ...           ...       ...   \n",
       "487    S423     Lgr5GFP       M5       GFPHigh      Stem   \n",
       "488    S424     Lgr5GFP       M5       GFPHigh      Stem   \n",
       "489    S425     Lgr5GFP       M5       GFPHigh      Stem   \n",
       "490    S427     Lgr5GFP       M5       GFPHigh      Stem   \n",
       "491    S429     Lgr5GFP       M5       GFPHigh      Stem   \n",
       "\n",
       "            full_cell_description Sample SRA Accession  \n",
       "0     S745_Lgr5GFP_M7_GFPLow_Stem   S745    SRX2419245  \n",
       "1     S856_Lgr5GFP_M7_GFPLow_Stem   S856    SRX2419356  \n",
       "2    S222_Lgr5GFP_M7_GFPHigh_Stem   S222    SRX2418722  \n",
       "3     S333_Lgr5GFP_M7_GFPLow_Stem   S333    SRX2418833  \n",
       "4     S589_Lgr5GFP_M7_GFPLow_Stem   S589    SRX2419089  \n",
       "..                            ...    ...           ...  \n",
       "487  S423_Lgr5GFP_M5_GFPHigh_Stem   S423    SRX2418923  \n",
       "488  S424_Lgr5GFP_M5_GFPHigh_Stem   S424    SRX2418924  \n",
       "489  S425_Lgr5GFP_M5_GFPHigh_Stem   S425    SRX2418925  \n",
       "490  S427_Lgr5GFP_M5_GFPHigh_Stem   S427    SRX2418927  \n",
       "491  S429_Lgr5GFP_M5_GFPHigh_Stem   S429    SRX2418929  \n",
       "\n",
       "[492 rows x 8 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'SRX2419104', 'SRX2418850', 'SRX2419603'}\n"
     ]
    }
   ],
   "source": [
    "countmatrix_rows = set(countmatrix.index)\n",
    "sra_accessions = set(adata.obs['SRA Accession'])\n",
    "missing_in_countmatrix_rows = sra_accessions - countmatrix_rows\n",
    "print(missing_in_countmatrix_rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "countmatrix['row_index'] = countmatrix.index\n",
    "merged_df = pd.merge(adata.obs, countmatrix, left_on='SRA Accession', right_on='row_index', how='inner')\n",
    "merged_df = merged_df.drop(columns=['row_index'])\n",
    "merged_df.set_index('SRA Accession', inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>barcode</th>\n",
       "      <th>Gene_marker</th>\n",
       "      <th>Donor_ID</th>\n",
       "      <th>GFP_intensity</th>\n",
       "      <th>Cell_Type</th>\n",
       "      <th>full_cell_description</th>\n",
       "      <th>Sample</th>\n",
       "      <th>ENSMUSG00000102693.2</th>\n",
       "      <th>ENSMUSG00000064842.3</th>\n",
       "      <th>ENSMUSG00000051951.6</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSMUSG00000096550.2</th>\n",
       "      <th>ENSMUSG00000094172.2</th>\n",
       "      <th>ENSMUSG00000094887.2</th>\n",
       "      <th>ENSMUSG00000091585.3</th>\n",
       "      <th>ENSMUSG00000095763.2</th>\n",
       "      <th>ENSMUSG00000095523.2</th>\n",
       "      <th>ENSMUSG00000095475.2</th>\n",
       "      <th>ENSMUSG00000094855.2</th>\n",
       "      <th>ENSMUSG00000095019.2</th>\n",
       "      <th>ENSMUSG00000095041.8</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRA Accession</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SRX2419245</th>\n",
       "      <td>S745</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S745_Lgr5GFP_M7_GFPLow_Stem</td>\n",
       "      <td>S745</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2419356</th>\n",
       "      <td>S856</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S856_Lgr5GFP_M7_GFPLow_Stem</td>\n",
       "      <td>S856</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418722</th>\n",
       "      <td>S222</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S222_Lgr5GFP_M7_GFPHigh_Stem</td>\n",
       "      <td>S222</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418833</th>\n",
       "      <td>S333</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S333_Lgr5GFP_M7_GFPLow_Stem</td>\n",
       "      <td>S333</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2419089</th>\n",
       "      <td>S589</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S589_Lgr5GFP_M7_GFPLow_Stem</td>\n",
       "      <td>S589</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418923</th>\n",
       "      <td>S423</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M5</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S423_Lgr5GFP_M5_GFPHigh_Stem</td>\n",
       "      <td>S423</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418924</th>\n",
       "      <td>S424</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M5</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S424_Lgr5GFP_M5_GFPHigh_Stem</td>\n",
       "      <td>S424</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>694</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418925</th>\n",
       "      <td>S425</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M5</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S425_Lgr5GFP_M5_GFPHigh_Stem</td>\n",
       "      <td>S425</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418927</th>\n",
       "      <td>S427</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M5</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S427_Lgr5GFP_M5_GFPHigh_Stem</td>\n",
       "      <td>S427</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418929</th>\n",
       "      <td>S429</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M5</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S429_Lgr5GFP_M5_GFPHigh_Stem</td>\n",
       "      <td>S429</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>605</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>489 rows × 57193 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              barcode Gene_marker Donor_ID GFP_intensity Cell_Type  \\\n",
       "SRA Accession                                                        \n",
       "SRX2419245       S745     Lgr5GFP       M7        GFPLow      Stem   \n",
       "SRX2419356       S856     Lgr5GFP       M7        GFPLow      Stem   \n",
       "SRX2418722       S222     Lgr5GFP       M7       GFPHigh      Stem   \n",
       "SRX2418833       S333     Lgr5GFP       M7        GFPLow      Stem   \n",
       "SRX2419089       S589     Lgr5GFP       M7        GFPLow      Stem   \n",
       "...               ...         ...      ...           ...       ...   \n",
       "SRX2418923       S423     Lgr5GFP       M5       GFPHigh      Stem   \n",
       "SRX2418924       S424     Lgr5GFP       M5       GFPHigh      Stem   \n",
       "SRX2418925       S425     Lgr5GFP       M5       GFPHigh      Stem   \n",
       "SRX2418927       S427     Lgr5GFP       M5       GFPHigh      Stem   \n",
       "SRX2418929       S429     Lgr5GFP       M5       GFPHigh      Stem   \n",
       "\n",
       "                      full_cell_description Sample ENSMUSG00000102693.2  \\\n",
       "SRA Accession                                                             \n",
       "SRX2419245      S745_Lgr5GFP_M7_GFPLow_Stem   S745                    0   \n",
       "SRX2419356      S856_Lgr5GFP_M7_GFPLow_Stem   S856                    0   \n",
       "SRX2418722     S222_Lgr5GFP_M7_GFPHigh_Stem   S222                    0   \n",
       "SRX2418833      S333_Lgr5GFP_M7_GFPLow_Stem   S333                    0   \n",
       "SRX2419089      S589_Lgr5GFP_M7_GFPLow_Stem   S589                    0   \n",
       "...                                     ...    ...                  ...   \n",
       "SRX2418923     S423_Lgr5GFP_M5_GFPHigh_Stem   S423                    0   \n",
       "SRX2418924     S424_Lgr5GFP_M5_GFPHigh_Stem   S424                    0   \n",
       "SRX2418925     S425_Lgr5GFP_M5_GFPHigh_Stem   S425                    0   \n",
       "SRX2418927     S427_Lgr5GFP_M5_GFPHigh_Stem   S427                    0   \n",
       "SRX2418929     S429_Lgr5GFP_M5_GFPHigh_Stem   S429                    0   \n",
       "\n",
       "              ENSMUSG00000064842.3 ENSMUSG00000051951.6  ...  \\\n",
       "SRA Accession                                            ...   \n",
       "SRX2419245                       0                    0  ...   \n",
       "SRX2419356                       0                    0  ...   \n",
       "SRX2418722                       0                    0  ...   \n",
       "SRX2418833                       0                    0  ...   \n",
       "SRX2419089                       0                    0  ...   \n",
       "...                            ...                  ...  ...   \n",
       "SRX2418923                       0                    0  ...   \n",
       "SRX2418924                       0                    0  ...   \n",
       "SRX2418925                       0                    0  ...   \n",
       "SRX2418927                       0                    0  ...   \n",
       "SRX2418929                       0                    0  ...   \n",
       "\n",
       "              ENSMUSG00000096550.2 ENSMUSG00000094172.2 ENSMUSG00000094887.2  \\\n",
       "SRA Accession                                                                  \n",
       "SRX2419245                       0                    0                    0   \n",
       "SRX2419356                       0                    0                    0   \n",
       "SRX2418722                       0                    0                    0   \n",
       "SRX2418833                       0                    0                    0   \n",
       "SRX2419089                       0                    0                    0   \n",
       "...                            ...                  ...                  ...   \n",
       "SRX2418923                       0                    0                    0   \n",
       "SRX2418924                       0                    0                    0   \n",
       "SRX2418925                       0                    0                    0   \n",
       "SRX2418927                       0                    0                    0   \n",
       "SRX2418929                       0                    0                    0   \n",
       "\n",
       "              ENSMUSG00000091585.3 ENSMUSG00000095763.2 ENSMUSG00000095523.2  \\\n",
       "SRA Accession                                                                  \n",
       "SRX2419245                       0                    0                    0   \n",
       "SRX2419356                       0                    0                    0   \n",
       "SRX2418722                       0                    0                    0   \n",
       "SRX2418833                       0                    0                    0   \n",
       "SRX2419089                       0                    0                    0   \n",
       "...                            ...                  ...                  ...   \n",
       "SRX2418923                       0                    0                    0   \n",
       "SRX2418924                       0                    0                    0   \n",
       "SRX2418925                       0                    0                    0   \n",
       "SRX2418927                       0                    0                    0   \n",
       "SRX2418929                       0                    0                    0   \n",
       "\n",
       "              ENSMUSG00000095475.2 ENSMUSG00000094855.2 ENSMUSG00000095019.2  \\\n",
       "SRA Accession                                                                  \n",
       "SRX2419245                       0                    0                    0   \n",
       "SRX2419356                       0                    0                    0   \n",
       "SRX2418722                       0                    0                    0   \n",
       "SRX2418833                       0                    0                    0   \n",
       "SRX2419089                       0                    0                    0   \n",
       "...                            ...                  ...                  ...   \n",
       "SRX2418923                       0                    0                    0   \n",
       "SRX2418924                       0                    0                    0   \n",
       "SRX2418925                       0                    0                    0   \n",
       "SRX2418927                       0                    0                    0   \n",
       "SRX2418929                       0                    0                    0   \n",
       "\n",
       "              ENSMUSG00000095041.8  \n",
       "SRA Accession                       \n",
       "SRX2419245                     583  \n",
       "SRX2419356                     274  \n",
       "SRX2418722                      11  \n",
       "SRX2418833                     153  \n",
       "SRX2419089                       2  \n",
       "...                            ...  \n",
       "SRX2418923                     378  \n",
       "SRX2418924                     694  \n",
       "SRX2418925                     391  \n",
       "SRX2418927                     398  \n",
       "SRX2418929                     605  \n",
       "\n",
       "[489 rows x 57193 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "obs = merged_df.iloc[:, :7]\n",
    "X = merged_df.iloc[:, 7:]\n",
    "var = pd.DataFrame(index=X.columns)\n",
    "new_adata = ad.AnnData(X=X.values, obs=obs, var=var)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>barcode</th>\n",
       "      <th>Gene_marker</th>\n",
       "      <th>Donor_ID</th>\n",
       "      <th>GFP_intensity</th>\n",
       "      <th>Cell_Type</th>\n",
       "      <th>full_cell_description</th>\n",
       "      <th>Sample</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRA Accession</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SRX2419245</th>\n",
       "      <td>S745</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S745_Lgr5GFP_M7_GFPLow_Stem</td>\n",
       "      <td>S745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2419356</th>\n",
       "      <td>S856</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S856_Lgr5GFP_M7_GFPLow_Stem</td>\n",
       "      <td>S856</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418722</th>\n",
       "      <td>S222</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S222_Lgr5GFP_M7_GFPHigh_Stem</td>\n",
       "      <td>S222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418833</th>\n",
       "      <td>S333</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S333_Lgr5GFP_M7_GFPLow_Stem</td>\n",
       "      <td>S333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2419089</th>\n",
       "      <td>S589</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M7</td>\n",
       "      <td>GFPLow</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S589_Lgr5GFP_M7_GFPLow_Stem</td>\n",
       "      <td>S589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418923</th>\n",
       "      <td>S423</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M5</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S423_Lgr5GFP_M5_GFPHigh_Stem</td>\n",
       "      <td>S423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418924</th>\n",
       "      <td>S424</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M5</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S424_Lgr5GFP_M5_GFPHigh_Stem</td>\n",
       "      <td>S424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418925</th>\n",
       "      <td>S425</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M5</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S425_Lgr5GFP_M5_GFPHigh_Stem</td>\n",
       "      <td>S425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418927</th>\n",
       "      <td>S427</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M5</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S427_Lgr5GFP_M5_GFPHigh_Stem</td>\n",
       "      <td>S427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRX2418929</th>\n",
       "      <td>S429</td>\n",
       "      <td>Lgr5GFP</td>\n",
       "      <td>M5</td>\n",
       "      <td>GFPHigh</td>\n",
       "      <td>Stem</td>\n",
       "      <td>S429_Lgr5GFP_M5_GFPHigh_Stem</td>\n",
       "      <td>S429</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>489 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              barcode Gene_marker Donor_ID GFP_intensity Cell_Type  \\\n",
       "SRA Accession                                                        \n",
       "SRX2419245       S745     Lgr5GFP       M7        GFPLow      Stem   \n",
       "SRX2419356       S856     Lgr5GFP       M7        GFPLow      Stem   \n",
       "SRX2418722       S222     Lgr5GFP       M7       GFPHigh      Stem   \n",
       "SRX2418833       S333     Lgr5GFP       M7        GFPLow      Stem   \n",
       "SRX2419089       S589     Lgr5GFP       M7        GFPLow      Stem   \n",
       "...               ...         ...      ...           ...       ...   \n",
       "SRX2418923       S423     Lgr5GFP       M5       GFPHigh      Stem   \n",
       "SRX2418924       S424     Lgr5GFP       M5       GFPHigh      Stem   \n",
       "SRX2418925       S425     Lgr5GFP       M5       GFPHigh      Stem   \n",
       "SRX2418927       S427     Lgr5GFP       M5       GFPHigh      Stem   \n",
       "SRX2418929       S429     Lgr5GFP       M5       GFPHigh      Stem   \n",
       "\n",
       "                      full_cell_description Sample  \n",
       "SRA Accession                                       \n",
       "SRX2419245      S745_Lgr5GFP_M7_GFPLow_Stem   S745  \n",
       "SRX2419356      S856_Lgr5GFP_M7_GFPLow_Stem   S856  \n",
       "SRX2418722     S222_Lgr5GFP_M7_GFPHigh_Stem   S222  \n",
       "SRX2418833      S333_Lgr5GFP_M7_GFPLow_Stem   S333  \n",
       "SRX2419089      S589_Lgr5GFP_M7_GFPLow_Stem   S589  \n",
       "...                                     ...    ...  \n",
       "SRX2418923     S423_Lgr5GFP_M5_GFPHigh_Stem   S423  \n",
       "SRX2418924     S424_Lgr5GFP_M5_GFPHigh_Stem   S424  \n",
       "SRX2418925     S425_Lgr5GFP_M5_GFPHigh_Stem   S425  \n",
       "SRX2418927     S427_Lgr5GFP_M5_GFPHigh_Stem   S427  \n",
       "SRX2418929     S429_Lgr5GFP_M5_GFPHigh_Stem   S429  \n",
       "\n",
       "[489 rows x 7 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_adata.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_adata.var = new_adata.var.astype(str)\n",
    "new_adata.var.index = new_adata.var.index.astype(str)\n",
    "new_adata.obs = new_adata.obs.astype(str)\n",
    "new_adata.obs.index = new_adata.obs.index.astype(str)\n",
    "new_adata.X = new_adata.X.astype(float)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_adata.write_h5ad('/../../../gut_project/Processed_data/Gut_data/Haber_2017_Smartseq/Haber_2017_Smartseq_stem_cells_remapped.h5ad')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "+ Add human paralogues"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_mouse_orthologues = pd.read_csv('/../../../ensembl_data/mouse_to_human_orthologues_ensembl.txt', sep='\\t', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "85346"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "duplicates = human_mouse_orthologues['Gene stable ID version'].duplicated()\n",
    "num_duplicates = duplicates.sum()\n",
    "num_duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_mouse_orthologues_cleaned = human_mouse_orthologues.loc[\n",
    "    human_mouse_orthologues.groupby('Gene stable ID version')['%id. target Human gene identical to query gene'].idxmax()\n",
    "]\n",
    "human_mouse_orthologues_cleaned = human_mouse_orthologues_cleaned.reset_index(drop=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_mouse_orthologues_cleaned = human_mouse_orthologues_cleaned.drop_duplicates(subset='Gene stable ID version')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20473"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "intersect = np.intersect1d(human_mouse_orthologues_cleaned['Gene stable ID version'], new_adata.var.index)\n",
    "len(intersect)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/amaguza/miniforge3/envs/scanpy_env/lib/python3.11/site-packages/anndata/_core/anndata.py:1113: FutureWarning: is_categorical_dtype is deprecated and will be removed in a future version. Use isinstance(dtype, CategoricalDtype) instead\n",
      "  if not is_categorical_dtype(df_full[k]):\n"
     ]
    }
   ],
   "source": [
    "human_mouse_orthologues_cleaned = human_mouse_orthologues_cleaned.loc[human_mouse_orthologues_cleaned['Gene stable ID version'].isin(intersect)]\n",
    "new_adata = new_adata[:, intersect]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_17745/1126677715.py:1: ImplicitModificationWarning: Trying to modify attribute `.var` of view, initializing view as actual.\n",
      "  new_adata.var['original_mouse_gene_id'] = new_adata.var.index.copy()\n"
     ]
    }
   ],
   "source": [
    "new_adata.var['original_mouse_gene_id'] = new_adata.var.index.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/amaguza/miniforge3/envs/scanpy_env/lib/python3.11/site-packages/anndata/_core/anndata.py:788: UserWarning: \n",
      "AnnData expects .var.index to contain strings, but got values like:\n",
      "    [0, 1, 5, 14, 18]\n",
      "\n",
      "    Inferred to be: integer\n",
      "\n",
      "  value_idx = self._prep_dim_index(value.index, attr)\n"
     ]
    }
   ],
   "source": [
    "new_adata.var = new_adata.var.merge(human_mouse_orthologues_cleaned, left_index=True, right_on='Gene stable ID version')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_adata.var = new_adata.var.set_index('Human gene name')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_adata.var.index = new_adata.var.index.astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>original_mouse_gene_id</th>\n",
       "      <th>Gene stable ID version</th>\n",
       "      <th>Transcript stable ID</th>\n",
       "      <th>Transcript stable ID version</th>\n",
       "      <th>Human gene stable ID</th>\n",
       "      <th>Human homology type</th>\n",
       "      <th>%id. target Human gene identical to query gene</th>\n",
       "      <th>%id. query gene identical to target Human gene</th>\n",
       "      <th>Human Gene-order conservation score</th>\n",
       "      <th>Human orthology confidence [0 low, 1 high]</th>\n",
       "      <th>Human protein or transcript stable ID</th>\n",
       "      <th>Query protein or transcript ID</th>\n",
       "      <th>Protein stable ID</th>\n",
       "      <th>Protein stable ID version</th>\n",
       "      <th>Gene name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Human gene name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GNAI3</th>\n",
       "      <td>ENSMUSG00000000001.5</td>\n",
       "      <td>ENSMUSG00000000001.5</td>\n",
       "      <td>ENSMUST00000000001</td>\n",
       "      <td>ENSMUST00000000001.5</td>\n",
       "      <td>ENSG00000065135</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>98.3051</td>\n",
       "      <td>98.3051</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSP00000358867</td>\n",
       "      <td>ENSMUSP00000000001</td>\n",
       "      <td>ENSMUSP00000000001</td>\n",
       "      <td>ENSMUSP00000000001.5</td>\n",
       "      <td>Gnai3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDC45</th>\n",
       "      <td>ENSMUSG00000000028.16</td>\n",
       "      <td>ENSMUSG00000000028.16</td>\n",
       "      <td>ENSMUST00000000028</td>\n",
       "      <td>ENSMUST00000000028.14</td>\n",
       "      <td>ENSG00000093009</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>91.3428</td>\n",
       "      <td>91.3428</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSP00000263201</td>\n",
       "      <td>ENSMUSP00000000028</td>\n",
       "      <td>ENSMUSP00000000028</td>\n",
       "      <td>ENSMUSP00000000028.8</td>\n",
       "      <td>Cdc45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SCML2</th>\n",
       "      <td>ENSMUSG00000000037.18</td>\n",
       "      <td>ENSMUSG00000000037.18</td>\n",
       "      <td>ENSMUST00000238603</td>\n",
       "      <td>ENSMUST00000238603.2</td>\n",
       "      <td>ENSG00000102098</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>36.5325</td>\n",
       "      <td>50.5714</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSP00000251900</td>\n",
       "      <td>ENSMUSP00000076593</td>\n",
       "      <td>ENSMUSP00000158772</td>\n",
       "      <td>ENSMUSP00000158772.2</td>\n",
       "      <td>Scml2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>APOH</th>\n",
       "      <td>ENSMUSG00000000049.12</td>\n",
       "      <td>ENSMUSG00000000049.12</td>\n",
       "      <td>ENSMUST00000000049</td>\n",
       "      <td>ENSMUST00000000049.6</td>\n",
       "      <td>ENSG00000091583</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>76.5217</td>\n",
       "      <td>76.5217</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSP00000205948</td>\n",
       "      <td>ENSMUSP00000000049</td>\n",
       "      <td>ENSMUSP00000000049</td>\n",
       "      <td>ENSMUSP00000000049.6</td>\n",
       "      <td>Apoh</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NARF</th>\n",
       "      <td>ENSMUSG00000000056.8</td>\n",
       "      <td>ENSMUSG00000000056.8</td>\n",
       "      <td>ENSMUST00000103015</td>\n",
       "      <td>ENSMUST00000103015.4</td>\n",
       "      <td>ENSG00000141562</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>84.4156</td>\n",
       "      <td>85.5263</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSP00000309899</td>\n",
       "      <td>ENSMUSP00000099304</td>\n",
       "      <td>ENSMUSP00000099304</td>\n",
       "      <td>ENSMUSP00000099304.4</td>\n",
       "      <td>Narf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RNU6-977P</th>\n",
       "      <td>ENSMUSG00002076963.1</td>\n",
       "      <td>ENSMUSG00002076963.1</td>\n",
       "      <td>ENSMUST00020183153</td>\n",
       "      <td>ENSMUST00020183153.1</td>\n",
       "      <td>ENSG00000252220</td>\n",
       "      <td>ortholog_one2many</td>\n",
       "      <td>60.3774</td>\n",
       "      <td>62.1359</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>ENST00000516411</td>\n",
       "      <td>ENSMUST00020183153</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gm54684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNORD100</th>\n",
       "      <td>ENSMUSG00002076971.1</td>\n",
       "      <td>ENSMUSG00002076971.1</td>\n",
       "      <td>ENSMUST00020182284</td>\n",
       "      <td>ENSMUST00020182284.1</td>\n",
       "      <td>ENSG00000221500</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>83.7838</td>\n",
       "      <td>81.5789</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>ENST00000408573</td>\n",
       "      <td>ENSMUST00020182284</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Snord100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5S_rRNA</th>\n",
       "      <td>ENSMUSG00002076988.1</td>\n",
       "      <td>ENSMUSG00002076988.1</td>\n",
       "      <td>ENSMUST00020182589</td>\n",
       "      <td>ENSMUST00020182589.1</td>\n",
       "      <td>ENSG00000277411</td>\n",
       "      <td>ortholog_one2many</td>\n",
       "      <td>42.3077</td>\n",
       "      <td>51.8868</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>ENST00000614916</td>\n",
       "      <td>ENSMUST00020182589</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gm56371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nan</th>\n",
       "      <td>ENSMUSG00002076990.1</td>\n",
       "      <td>ENSMUSG00002076990.1</td>\n",
       "      <td>ENSMUST00020183326</td>\n",
       "      <td>ENSMUST00020183326.1</td>\n",
       "      <td>ENSG00000281311</td>\n",
       "      <td>ortholog_many2many</td>\n",
       "      <td>79.2593</td>\n",
       "      <td>79.2593</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>ENST00000628590</td>\n",
       "      <td>ENSMUST00020183326</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gm22711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RN7SKP183</th>\n",
       "      <td>ENSMUSG00002076992.1</td>\n",
       "      <td>ENSMUSG00002076992.1</td>\n",
       "      <td>ENSMUST00020181762</td>\n",
       "      <td>ENSMUST00020181762.1</td>\n",
       "      <td>ENSG00000252978</td>\n",
       "      <td>ortholog_one2one</td>\n",
       "      <td>39.7394</td>\n",
       "      <td>37.6543</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>ENST00000517169</td>\n",
       "      <td>ENSMUST00020181762</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gm54807</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20473 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                original_mouse_gene_id Gene stable ID version  \\\n",
       "Human gene name                                                 \n",
       "GNAI3             ENSMUSG00000000001.5   ENSMUSG00000000001.5   \n",
       "CDC45            ENSMUSG00000000028.16  ENSMUSG00000000028.16   \n",
       "SCML2            ENSMUSG00000000037.18  ENSMUSG00000000037.18   \n",
       "APOH             ENSMUSG00000000049.12  ENSMUSG00000000049.12   \n",
       "NARF              ENSMUSG00000000056.8   ENSMUSG00000000056.8   \n",
       "...                                ...                    ...   \n",
       "RNU6-977P         ENSMUSG00002076963.1   ENSMUSG00002076963.1   \n",
       "SNORD100          ENSMUSG00002076971.1   ENSMUSG00002076971.1   \n",
       "5S_rRNA           ENSMUSG00002076988.1   ENSMUSG00002076988.1   \n",
       "nan               ENSMUSG00002076990.1   ENSMUSG00002076990.1   \n",
       "RN7SKP183         ENSMUSG00002076992.1   ENSMUSG00002076992.1   \n",
       "\n",
       "                Transcript stable ID Transcript stable ID version  \\\n",
       "Human gene name                                                     \n",
       "GNAI3             ENSMUST00000000001         ENSMUST00000000001.5   \n",
       "CDC45             ENSMUST00000000028        ENSMUST00000000028.14   \n",
       "SCML2             ENSMUST00000238603         ENSMUST00000238603.2   \n",
       "APOH              ENSMUST00000000049         ENSMUST00000000049.6   \n",
       "NARF              ENSMUST00000103015         ENSMUST00000103015.4   \n",
       "...                              ...                          ...   \n",
       "RNU6-977P         ENSMUST00020183153         ENSMUST00020183153.1   \n",
       "SNORD100          ENSMUST00020182284         ENSMUST00020182284.1   \n",
       "5S_rRNA           ENSMUST00020182589         ENSMUST00020182589.1   \n",
       "nan               ENSMUST00020183326         ENSMUST00020183326.1   \n",
       "RN7SKP183         ENSMUST00020181762         ENSMUST00020181762.1   \n",
       "\n",
       "                Human gene stable ID Human homology type  \\\n",
       "Human gene name                                            \n",
       "GNAI3                ENSG00000065135    ortholog_one2one   \n",
       "CDC45                ENSG00000093009    ortholog_one2one   \n",
       "SCML2                ENSG00000102098    ortholog_one2one   \n",
       "APOH                 ENSG00000091583    ortholog_one2one   \n",
       "NARF                 ENSG00000141562    ortholog_one2one   \n",
       "...                              ...                 ...   \n",
       "RNU6-977P            ENSG00000252220   ortholog_one2many   \n",
       "SNORD100             ENSG00000221500    ortholog_one2one   \n",
       "5S_rRNA              ENSG00000277411   ortholog_one2many   \n",
       "nan                  ENSG00000281311  ortholog_many2many   \n",
       "RN7SKP183            ENSG00000252978    ortholog_one2one   \n",
       "\n",
       "                 %id. target Human gene identical to query gene  \\\n",
       "Human gene name                                                   \n",
       "GNAI3                                                   98.3051   \n",
       "CDC45                                                   91.3428   \n",
       "SCML2                                                   36.5325   \n",
       "APOH                                                    76.5217   \n",
       "NARF                                                    84.4156   \n",
       "...                                                         ...   \n",
       "RNU6-977P                                               60.3774   \n",
       "SNORD100                                                83.7838   \n",
       "5S_rRNA                                                 42.3077   \n",
       "nan                                                     79.2593   \n",
       "RN7SKP183                                               39.7394   \n",
       "\n",
       "                 %id. query gene identical to target Human gene  \\\n",
       "Human gene name                                                   \n",
       "GNAI3                                                   98.3051   \n",
       "CDC45                                                   91.3428   \n",
       "SCML2                                                   50.5714   \n",
       "APOH                                                    76.5217   \n",
       "NARF                                                    85.5263   \n",
       "...                                                         ...   \n",
       "RNU6-977P                                               62.1359   \n",
       "SNORD100                                                81.5789   \n",
       "5S_rRNA                                                 51.8868   \n",
       "nan                                                     79.2593   \n",
       "RN7SKP183                                               37.6543   \n",
       "\n",
       "                 Human Gene-order conservation score  \\\n",
       "Human gene name                                        \n",
       "GNAI3                                          100.0   \n",
       "CDC45                                          100.0   \n",
       "SCML2                                          100.0   \n",
       "APOH                                           100.0   \n",
       "NARF                                           100.0   \n",
       "...                                              ...   \n",
       "RNU6-977P                                        NaN   \n",
       "SNORD100                                         NaN   \n",
       "5S_rRNA                                          NaN   \n",
       "nan                                              NaN   \n",
       "RN7SKP183                                        NaN   \n",
       "\n",
       "                 Human orthology confidence [0 low, 1 high]  \\\n",
       "Human gene name                                               \n",
       "GNAI3                                                     1   \n",
       "CDC45                                                     1   \n",
       "SCML2                                                     0   \n",
       "APOH                                                      1   \n",
       "NARF                                                      1   \n",
       "...                                                     ...   \n",
       "RNU6-977P                                                 0   \n",
       "SNORD100                                                  1   \n",
       "5S_rRNA                                                   0   \n",
       "nan                                                       1   \n",
       "RN7SKP183                                                 0   \n",
       "\n",
       "                Human protein or transcript stable ID  \\\n",
       "Human gene name                                         \n",
       "GNAI3                                 ENSP00000358867   \n",
       "CDC45                                 ENSP00000263201   \n",
       "SCML2                                 ENSP00000251900   \n",
       "APOH                                  ENSP00000205948   \n",
       "NARF                                  ENSP00000309899   \n",
       "...                                               ...   \n",
       "RNU6-977P                             ENST00000516411   \n",
       "SNORD100                              ENST00000408573   \n",
       "5S_rRNA                               ENST00000614916   \n",
       "nan                                   ENST00000628590   \n",
       "RN7SKP183                             ENST00000517169   \n",
       "\n",
       "                Query protein or transcript ID   Protein stable ID  \\\n",
       "Human gene name                                                      \n",
       "GNAI3                       ENSMUSP00000000001  ENSMUSP00000000001   \n",
       "CDC45                       ENSMUSP00000000028  ENSMUSP00000000028   \n",
       "SCML2                       ENSMUSP00000076593  ENSMUSP00000158772   \n",
       "APOH                        ENSMUSP00000000049  ENSMUSP00000000049   \n",
       "NARF                        ENSMUSP00000099304  ENSMUSP00000099304   \n",
       "...                                        ...                 ...   \n",
       "RNU6-977P                   ENSMUST00020183153                 NaN   \n",
       "SNORD100                    ENSMUST00020182284                 NaN   \n",
       "5S_rRNA                     ENSMUST00020182589                 NaN   \n",
       "nan                         ENSMUST00020183326                 NaN   \n",
       "RN7SKP183                   ENSMUST00020181762                 NaN   \n",
       "\n",
       "                Protein stable ID version Gene name  \n",
       "Human gene name                                      \n",
       "GNAI3                ENSMUSP00000000001.5     Gnai3  \n",
       "CDC45                ENSMUSP00000000028.8     Cdc45  \n",
       "SCML2                ENSMUSP00000158772.2     Scml2  \n",
       "APOH                 ENSMUSP00000000049.6      Apoh  \n",
       "NARF                 ENSMUSP00000099304.4      Narf  \n",
       "...                                   ...       ...  \n",
       "RNU6-977P                             NaN   Gm54684  \n",
       "SNORD100                              NaN  Snord100  \n",
       "5S_rRNA                               NaN   Gm56371  \n",
       "nan                                   NaN   Gm22711  \n",
       "RN7SKP183                             NaN   Gm54807  \n",
       "\n",
       "[20473 rows x 15 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_adata.var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_adata.write('/../../../gut_project/Processed_data/Gut_data/Haber_2017_Smartseq/Haber_2017_Smartseq_stem_cells_remapped_human_orthologues_genes.h5ad')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Scanpy",
   "language": "python",
   "name": "scnapy-server"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
